app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CompleteReturn.scala

```diff
@@ -81,7 +82,8 @@ object CompleteReturn {
               u,
               maybeCompleteRepresenteeAnswers,
               gainOrLossAfterReliefs,
-              hasAttachments = true
+              hasAttachments = true,
+              isFirstReturn = true
```

Comment:
a draft return is not always necessarily a first return so this value is incorrect. I originally suggested sticking this `isFirstReturn` flag in the `ViewingReturn` journey status to avoid having to set it here. While we're filling our a return, that information is already contained within `FillingOutReturn`, so there'd be a duplication of information sticking it here.

Reply:
If see the BE draft PR, the flag is set there in CompleteReturn  and what I did was to map that type for type in the FE. Is that not correct? Because I did that the problem above occurs. What I think you are saying now is that once the response is returned from the BE, determine the flag and then set it in the `ViewingReturn` and not in the `CompleteReturn` subtypes?

Reply:
ah ok, I can see it being set in the BE now when reading back the return from DES. Putting the flag in `CompleteReturn` though means that you also need to send it in the submit return request from the FE to the BE which is why you're having to populate the flag here in the `apply` methods that take you from a draft return to a complete return. Even though it's not being read by the BE in the submit return request it's still there together with the actual `isFurtherReturn: Boolean` parameter you've put in the submit return request. Sticking it outside means that you can avoid having to do that but it means you'll have to update the response to get a return to include the `isFirstReturn` flag outside of `CompleteReturn` which would mean updating the JSON structure of the response.

Reply:
the way forward with the least amount of changes then would be to add in an extra parameter to these apply methods to give the proper value of `isFirstReturn`, which would be equal to `!isFurtherReturn` which should be available from the place they are being called.

Reply:
it seems cleaner to stick it outside though so you don't have to do the extra work of populating it when you don't need it

Reply:
I think what you're saying now is that instead of augmenting `CompleteReturn` in the BE with the flag, return a new type which is a composite of the `CompleteReturn` and the flag. That way the subtypes of `CompleteReturn` don't need to be changed in the BE and neither in the FE.

Reply:
yeah, that's basically it just in a lot fewer words :smile:

Reply:
added a new type called `DisplayReturn` - can you re-review the FE and BE draft PRs. If fine, then I will fix the FE broken tests.

in particular can you check the changes around the new `StartingAmendReturn` type you added yesterday

Reply:
@AndyHWChung :point_up:

Reply:
changes look good - it's good to have the new field somewhere in `StartingAmendReturn` - we're gonna need that flag for something very soon

Reply:
added flag to `StartingAmendReturn`